07/fundamentals.html

```diff
@@ -185,33 +186,32 @@ <h2 id="flushing"><a href="#flushing">Message queueing, chunks and flushing</a><
     }
             </code>
         </pre>
-        And <code>m_Connection.QueueChunk()</code> as the name suggests only puts
-        these messages into a queue of so called chunks. So no packet header is created
+        As the name suggests, <code>m_Connection.QueueChunk()</code> only puts
+        these messages into a <em>queue</em> of so-called chunks; no packet header is created
         and no data is being sent over the network yet. For now the messages only get
         collected in a queue.
         This queue is then sent as soon as it is full or a message with the flag
-        <code>MSGFLAG_FLUSH</code> is being send. When it is being send
+        <code>MSGFLAG_FLUSH</code> is sent. When <code>MSGFLAG_FLUSH</code> is sent,
         all messages in the queue get their own chunk header and in the very beginning of
-        the udp packet there is the packet header which also includes one field that says
-        how many chunks are in the packet.
+        the UDP packet, the packet header includes one field stating the amount of chunks in the packet.
         <br>
         <br>
-        This is probably done for performance reasons to not waste bandwith on teeworlds/udp/ip headers
-        when non time critical messages are being sent that are followed by other messages anyways.
-        So they are bundled up and sent together.
+        This is probably done for performance reasons: to avoid wasting bandwidth on Teeworlds/UDP headers
```

Comment:
why is ip gone here? I wanted to mention that the thing is wrapped in 3 protocols and all of them bring a header.

Reply:
Oh ok I thought it was just cruft that didn't add much to the content, will add it back